pkg/kube/configmap/configmap.go

```diff
@@ -85,14 +85,14 @@ func UpdateField(ctx context.Context, getUpdater GetUpdater, objectKey client.Ob
 // CreateOrUpdate creates the given ConfigMap if it doesn't exist,
 // or updates it if it does.
 func CreateOrUpdate(ctx context.Context, getUpdateCreator GetUpdateCreator, cm corev1.ConfigMap) error {
-	_, err := getUpdateCreator.GetConfigMap(ctx, types.NamespacedName{Name: cm.Name, Namespace: cm.Namespace})
-	if err != nil {
+	if err := getUpdateCreator.UpdateConfigMap(ctx, cm); err != nil {
 		if apiErrors.IsNotFound(err) {
 			return getUpdateCreator.CreateConfigMap(ctx, cm)
+		} else {
```

Comment:
i am a bit late to the party but we could have saved the else and just return, right?

Reply:
When I see such if/else I always wonder if I should leave a comment
Should we try to be consistent in the whole codebase about that ?
We use both currently